addons/forum-search/userscript.js

```diff
@@ -56,30 +55,41 @@ function cleanPost(post) {
   return readableDom.body.cloneNode(true).childNodes;
 }
 
-function triggerNewSearch(searchContent, query, sort, msg) {
+function triggerNewSearch(searchContent, query, filter, sort, msg) {
   searchContent.classList.add("show");
   while (searchContent.firstChild) {
     searchContent.removeChild(searchContent.firstChild);
   }
   currentQuery = query;
-  appendSearch(searchContent, query, 0, sort, msg);
+  appendSearch(searchContent, query, filter, 0, sort, msg);
 }
 
-function appendSearch(box, query, page, term, msg) {
+function appendSearch(box, query, filter, page, term, msg) {
   if (page * 50 > hits) return 0;
   isCurrentlyProcessing = true;
   let loading = document.createTextNode(msg("loading"));
   currentPage = page;
   box.appendChild(loading);
   window
-    .fetch(`https://scratchdb.lefty.one/v3/forum/search?q=${encodeURIComponent(query)}&page=${page}&o=${term}`)
+    .fetch(
+      `https://scratchdb.lefty.one/search/indexes/forum_posts/search?attributesToSearchOn=content&hitsPerPage=50&q=${encodeURIComponent(
+        query
+      )}&filter=${encodeURIComponent(filter)}&page=${page + 1}${
+        term == "newest" ? "&sort=id:desc" : term == "oldest" ? "&sort=id:asc" : ""
+      }`,
+      {
+        headers: {
+          authorization: "Bearer 3396f61ef5b02abf801096be5f0b0ee620de304dd92fc6045aeb99539cd0bec4",
```

Comment:
Adding note about this token being public

```suggestion
          // Note: the following token is well-known and public.
          authorization: "Bearer 3396f61ef5b02abf801096be5f0b0ee620de304dd92fc6045aeb99539cd0bec4",
```